discovery-provider/src/tasks/index_solana_plays.py

```diff
@@ -402,7 +402,10 @@ def parse_sol_tx_batch(
     # This means every subsequent run will continue to no-op on error transactions and the cache will never be updated
     if tx_sig_batch_records and not plays:
         logger.info("index_solana_plays.py | Clearing redis cache")
-        redis.delete(REDIS_TX_CACHE_QUEUE_PREFIX)
+        # redis.delete(REDIS_TX_CACHE_QUEUE_PREFIX)
+
+        # don't delete instead, trim
+        redis.ltrim(REDIS_TX_CACHE_QUEUE_PREFIX, 1, -1)
```

Comment:
this is a good idea - but shouldn't we trim >1? as in - remove until a valid one is found?

Reply:
probably need to cache whether its an error or not too

Reply:
can we skip caching for errors?

Reply:
this might be too much of an edge case to even matter but i was just hoping to optimize the case:
error, error, ...error, 3, 4, 5,... 10000.

by deleting this whole queue, we'd have to go backwards and try catch up rather than just progressing forwards. so yeah maybe it should be like 1000 rather than 1?